data_structures/binary_tree/binary_tree_maximum_path_sum.py

```diff
@@ -0,0 +1,132 @@
+"""
+A path in a binary tree is a sequence of nodes where each pair
+of adjacent nodes in the sequence has an edge connecting
+them. A node can only appear in the sequence at most once. Note
+that the path does not need to pass through the root.
+
+The path sum of a path is the sum of the node's values in the path.
+
+Given the root of a binary tree, return the maximum path sum of any non-empty path.
+
+Leetcode Reference : https://leetcode.com/problems/binary-tree-maximum-path-sum/
+"""
+
+
+class TreeNode:
+
+    """
+    TreeNode has tree variables, val -> Stores value of the node
+    left, right -> Stores the pointer to left or right node.
+    """
+
+    def __init__(self, val: int) -> None:
+        self.val: int = val
+        self.left: TreeNode | None = None
+        self.right: TreeNode | None = None
+
+
+class GetMaxPathSum:
```

Comment:
This is a _custom_ class only designed the get the sum.

I would prefer to see a _generic_ `BinaryTree` class that:
1. has a `self.root_node`
2. implements `__iter__()` instead of travers.
3. delivers the solution with the simple line `sum(binary_tree)`

Reply:
Like this but for a BinaryTree instead of a LinkedList.
https://github.com/TheAlgorithms/Python/blob/master/data_structures/linked_list/reverse_k_group.py#L13

Reply:
Perhaps 3. is not possible for this problem.

Reply:
@cclauss I'm  bit confused about __iter__(), as traverse recursively takes maximum from left and right subtree, how can I implement that. Please give me hint on it.

Reply:
Yes.  Now I see the difficulties.  Please make the BinaryTree and make a `.traverse()` method that yields the values.

Reply:
@cclauss As the algorithm traverses all the nodes recursively, if we try to yield values of nodes, we'll end up yielding all the nodes. Because yield will go over all nodes where's our algorithm only requires those nodes that makes up the maximum value. Please correct me if i'm wrong.

Reply:
You are probably right.

Reply:
@cclauss Hello, My PR getting out of the list now. Can you hear me back please how can I improve more.